.buildkite/release.yml

```diff
@@ -2,6 +2,16 @@ agents:
   provider: "gcp"
 
 steps:
+  - uses: elastic/apm-pipeline-library/.github/actions/github-token@current
+    with:
+      url: ${{ secrets.VAULT_ADDR }}
+      roleId: ${{ secrets.VAULT_ROLE_ID }}
+      secretId: ${{ secrets.VAULT_SECRET_ID }}
+  - uses: elastic/apm-pipeline-library/.github/actions/setup-git@current
+    with:
+      username: ${{ env.GIT_USER }}
+      email: ${{ env.GIT_EMAIL }}
+      token: ${{ env.GITHUB_TOKEN }}
```

Comment:
Sorry if I created any confusion. 
However, this is GH workflow syntax. This won't work in a buildkite pipeline.

Reply:
Here is example on how secrets are currently acquired from vault.

https://github.com/elastic/apm-agent-android/blob/b42538638bd521297bf7a9f9f855150fc48ae3c7/.ci/release.sh#L62

You would need to get the token in the same manner.

Unfortunately, I don't know if there is a token for apmmachine stored in the ci vault.

@elastic/observablt-ci anyone of you knows how to get the secret for `apmmachine` in buildkite?

Reply:
Oh! I see, tbh I thought it seemed too easy to be true 😅 - I've been mostly looking at how this has been done in the Java Agent repo. And after a closer look, it seems like the code change pushed into main is done outside of Buildkite, so I'm thinking that maybe that can be done here as well, in case getting this token inside Buildkite might not be worth the trouble... So I'll take a look at that alternative in the meantime, cheers.

Reply:
> @elastic/observablt-ci anyone of you knows how to get the secret for `apmmachine` in buildkite?

Buildkite can retrieve secrets from another vault.
I will send the documentation by DM since it's internal.

Reply:
I just moved the post-deploy command to the GH release.yml file [here](https://github.com/elastic/apm-agent-android/pull/169/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34R69). Alongside the GIT auth set up as well. I think it should work. Any thoughts @reakaleek @amannocci ?

Reply:
Actually, thinking about it a bit further, I might have to pull the agent project in the Action before running that command.

Reply:
If I add another job in the GH release.yml file (there's currently only one, which is the `release` one) is it possible to configure that second job to run after the `release` one only if it succeeded?

Reply:
I've added a new GH action job that should run after the release has been successful. In this new job we should do the version bump and changelog update and it should all get pushed straight into the `main` branch. I think it should work the way I wrote it, if there are any issues please let me know @amannocci @reakaleek - Otherwise I'm planning to merge this by the end of this week and apply future adjustments if needed since this PR has been open for too long now.

Reply:
I think you need to set `waitFor` to `true`

https://github.com/elastic/apm-agent-android/blob/947996594fa182ce11f5cb4d93d76430b75a657d/.github/workflows/release.yml#L50

so this works as expected

Reply:
Another thing I noticed:

pushing to protected branches does not work out of the box in github actions.

You need to use a GH Personal Access Token that has elevated permissions.

Reply:
You can find an example job here with the necessary setup to do pushing to a protected branch:


https://github.com/elastic/apm-agent-java/blob/main/.github/workflows/release.yml#L128-L151


Also, we need to add @apmmachine as a collaborating with a role that is able to bypass branch protection.